lib/src/chart/bar_chart/bar_chart_painter.dart

```diff
@@ -347,7 +347,18 @@ class BarChartPainter extends AxisChartPainter<BarChartData> {
               final stackFromY = getPixelY(stackItem.fromY, drawSize, holder);
               final stackToY = getPixelY(stackItem.toY, drawSize, holder);
 
-              _barPaint.color = stackItem.color;
+              if (stackItem.colors.length == 1) {
+                _barPaint.color = stackItem.colors[0];
+                _barPaint.shader = null;
+              } else {
+                _barPaint.shader = ui.Gradient.linear(
+                  Offset(left, stackFromY),
```

Comment:
Is it drawing a skewed gradient from `bottolLeft` to `topRight`?
I think we need to change this with something like `bottomCenter` to `topCenter` (or `topCenter` to `bottomCenter` when value is negative).

What do you think?

Reply:
You are right! Sorry I am not very experienced)

Reply:
Yeah you are right! I will do as you said). Sorry for lack of experience/attention)

Reply:
No worries.
Contribution is an excellent step to learning and gaining experience.
We are here to help each other.

Reply:
Regarding bottom center/top center your are right).  But, gradient direction should not change even if it is negative, I guess. Because, bars are relative to x axis (0). 

Here is how it looks if we change direction when negative. 
![test](https://user-images.githubusercontent.com/50243010/156787869-97b79839-0618-4c43-9fae-372484d3a4e5.jpg)

Reply:
@SomeoneAndNoone, we just changed the bars (check #929). 
We now have `fromY` and `toY` concepts (instead of only `y').
That's why your branch now has some conflicts.

Then, I think we should apply this approach to our gradient logic:
The gradient goes through `fromY` to `toY` values. It means the direction is matters.
If it is downward, the gradient should be `from: topCenter` and `to: bottomCenter`.
If it is upward, the gradient should be `from: bottomCenter` and `to: topCenter`.

What do you think?